packages/noya-state/src/reducers/application.ts

```diff
@@ -791,16 +798,30 @@ export function reducer(
         const sharedStyle: Sketch.SharedStyle = {
           _class: 'sharedStyle',
           do_objectID: uuid(),
-          name: 'New Layer Style',
-          value: produce(Models.style, (style) => {
-            style.do_objectID = uuid();
-            return style;
-          }),
+          name: name === undefined || name === '' ? 'New Layer Style' : name,
+          value: produce(
+            style === undefined ? Models.style : style,
+            (style) => {
+              style.do_objectID = uuid();
+              return style;
+            },
+          ),
         };
 
         layerStyles.objects.push(sharedStyle);
         state.sketch.document.layerStyles = layerStyles;
-        state.selectedLayerStyleIds = [sharedStyle.do_objectID];
+
+        if (style === undefined) {
+          state.selectedLayerStyleIds = [sharedStyle.do_objectID];
+        } else {
+          accessPageLayers(state, pageIndex, layerIndexPaths).forEach(
+            (layer) => {
+              if (!layer.style) return;
```

Comment:
I don't think we need `if (!layer.style) return;`. If the style doesn't exist, we have a different problem 😂